include/bus.h

```diff
@@ -193,8 +196,7 @@ static inline resource_t *bus_alloc_resource_anywhere(device_t *dev,
 
 /*! \brief Allocates resource for a device.
  *
- * Basically the same as \sa bus_alloc_resource_anywhere, but resource
- * has to be identifiable by parent bus driver by \param rid.
+ * \sa bus_alloc_resource_anywhere using default resource size for the bus
```

Comment:
Currently it's used only by std_vga driver. Numbers passed as `rid` are numbers of BAR registers. Parent bus has identified BARs for this device and can map them in PCI MEMORY space (seem `malta.h`), and then return as a resource to std_vga device.
`rid` is not passed to `rman_alloc_resource` it is only used by parent bus to find resource by id.

Reply:
I don't see how your comment is related to the actual change and the source code 

This is how the bus_alloc_resource_any looks like:
```c
static inline resource_t *bus_alloc_resource_any(device_t *dev, res_type_t type,
                                                 int rid, res_flags_t flags) {

  return BUS_DRIVER(dev)->bus.alloc_resource(dev->parent, dev, type, rid, 0,
                                             RMAN_ADDR_MAX, 1, flags);
}
```
It does the very same as thing as `bus_alloc_resource_anywhere` - with one difference - it sets the size to the default value equal to 1. I don't want to have a leaky abstraction here and comment that some internal rman is using the rid for certain action

Reply:
Sometimes driver knows what resources it has and can ask for them using `rid`. It doesn't care what's the size of that resource. It that case the resource already exists and parentbus can look it up (using `rid`).
Probably the interface should be redesigned when we have devhints implemented.